core/meta/src/TStatusBitsChecker.cxx

```diff
@@ -0,0 +1,272 @@
+}
+
+void TStatusBitsChecker::Registry::RegisterBits(TClass &classRef /* = false */)
+{
+   TEnum *eStatusBits = (TEnum*)classRef.GetListOfEnums()->FindObject("EStatusBits");
+   TEnum *exceptionBits = (TEnum*)classRef.GetListOfEnums()->FindObject("EStatusBitsDupExceptions");
+
+   if (eStatusBits) {
+
+      for(auto c : *eStatusBits->GetConstants()) {
+         TEnumConstant *constant = dynamic_cast<TEnumConstant*>(c);
+
+         // Ignore the known/intentional duplication.
+         bool intentionalDup = exceptionBits && exceptionBits->GetConstant(constant->GetName());
+
+         auto value = constant->GetValue();
+         auto bit = ConvertToBit(value, classRef, constant->GetName());
+
+         if (bit < 24) {
+            bool need = true;
+            for(auto reg : fRegister[bit]) {
+               if (reg.fOwner == &classRef) {
+                  // We have a duplicate declared in the same class
+                  // let's accept this as an alias.
+                  need = false;
+               }
+            }
+
+            if (need)
+               fRegister[bit].emplace_back(classRef,
+                                             std::string(constant->GetName()),
+                                             intentionalDup);
+         }
+      }
+
+   }
+
+   TList* lb = classRef.GetListOfBases();
+   if (lb) {
+      for(auto b : *lb) {
+         TBaseClass* base = dynamic_cast<TBaseClass*>(b);
+
+         RegisterBits(*(base->GetClassPointer()));
+      }
+   }
+}
+
+
+// Return false if there is any unexpected duplicates BIT constant in
+// the class hierarchy.
+bool TStatusBitsChecker::Registry::Check(TClass &classRef, bool verbose /* = false */)
+{
+   RegisterBits(classRef);
+
+   if (verbose) {
+      for(auto cursor : fRegister) {
+         for(auto constant : cursor.second) {
+            Printf("Bit %3d declared in %s as %s",cursor.first,
+                     constant.fOwner->GetName(),
+                     constant.fConstantName.c_str());
+         }
+      }
+   }
+
+   bool issuedHeader = false;
+   bool result = true;
+   for(auto cursor : fRegister) {
+      unsigned int nDuplicate = 0;
+      for(auto constant : cursor.second) {
+         if (!constant.fIntentionalDup)
+            ++nDuplicate;
+      }
+      if (nDuplicate > 1) {
+         if (!issuedHeader) {
+            Error("TStatusBitsChecker","In %s class hierarchy, there are duplicates bits:",classRef.GetName());
+            issuedHeader = true;
+         }
+         for(auto constant : cursor.second) {
+            if (!constant.fIntentionalDup) {
+               Error("TStatusBitsChecker", "   Bit %3d used in %s as %s",
+                     cursor.first,
+                     constant.fOwner->GetName(),
+                     constant.fConstantName.c_str());
+               result = false;
+            }
+         }
+      }
+   }
+
+   return result;
+
+}
+
+// Return false if there is any unexpected duplicates BIT constant in
```

Comment:
Why isn't this a proper doxygen comment?

Reply:
What does doxygen do with this lame comment?

Reply:
humm ... actually what is our current strategy ... the official documentation is in the header files.  Should we leave the source file completely undocumented?

Reply:
ROOT's function doc is in the sources.